### PR TITLE
APM uninstrumentation

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
Allows for disabling tracing (and anything that depends on tracing, e.g. appsec) on the fly by extending remote config support for DD_TRACING_ENABLED

[RFC](https://docs.google.com/document/d/1LXsAdg7M_oCv0PBId7V0SU28qioAtAgZlTf8tnvIf4o/edit?usp=sharing)

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: packages/dd-trace/src/proxy.js [hunk 1]
```diff
@@ -121,6 +108,27 @@ class Tracer extends NoopProxy {
...
+      if (!this._tracingInitialized) {
+        this._tracer = new DatadogTracer(config)
+        this.appsec = new AppsecSdk(this._tracer, config)
+        this._tracingInitialized = true
+      }
+      if (config.iast.enabled) {
+        require('./appsec/iast').enable(config, this._tracer) // disable
+      }
+      this._pluginManager.configure(config)
+    } else {
```
Comment (rochdev) [inline packages/dd-trace/src/proxy.js:126]: This is missing any code needed to disable other things. While it's possible that updating the config object alone would work, it's best to be safe and explicitly reconfigure everything.

Comment (rochdev) [reply packages/dd-trace/src/proxy.js:126]: Basically, same as above. The `configure` calls on the tracer and the plugin manager should be happening in this function.

File: packages/dd-trace/src/span_processor.js [hunk 2]
```diff
@@ -27,10 +27,14 @@ class SpanProcessor {
     const active = []
     const formatted = []
     const trace = spanContext._trace
-    const { flushMinSpans } = this._config
+    const { flushMinSpans, tracing } = this._config
     const { started, finished } = trace
 
     if (trace.record === false) return
+    if (!tracing) { // TODO: add test
```
Comment (rochdev) [inline packages/dd-trace/src/span_processor.js:34]: Let's elaborate on this TODO.

File: packages/dd-trace/test/span_processor.spec.js [hunk 3]
```diff
@@ -42,6 +42,7 @@ describe('SpanProcessor', () => {
       sample: sinon.stub()
     }
     config = {
+      tracing: true,
```
Comment (rochdev) [inline packages/dd-trace/test/span_processor.spec.js:45]: Why is this needed if it's the default? Is there a bug in the new code? I wonder if it's related to the large changes in benchmark results.

File: packages/dd-trace/src/proxy.js [hunk 4]
```diff
@@ -121,6 +108,27 @@ class Tracer extends NoopProxy {
     return this
   }
 
+  _enableOrDisableTracing (config) { // TODO: add test
```
Comment (rochdev) [inline packages/dd-trace/src/proxy.js:111]: This should be tested. Enabling and disabling the tracer at runtime is a fairly big update and we should make sure it works properly.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/proxy.js (hunk 4 above)
Comment (ida613) [reply packages/dd-trace/src/proxy.js:110]: make sense, just added unit tests for it

**Review by rochdev — APPROVED** [review]
